### PR TITLE
Datetime - avoid onChange triggered on initial rendering

### LINKED ISSUE #145: Datetime is sending onChange event on initial render
The Datetime component is sending an unwanted onChange event during it's first render. Thus changing the initial value to a moment object.

https://github.com/strapi/strapi/issues/7045

### PR DESCRIPTION
**Pull request by ericliu-el**
Hi there, I made some changes on the Datetime component, hope it helps

resolve #145

### CODE REVIEW

File: packages/buffetjs-custom/src/components/DateTime/index.js [hunk 1]
```diff
@@ -78,9 +78,12 @@ function DateTime({
 
   useEffect(() => {
     if (!!value && moment(value).isValid()) {
-      const newDate = value._isAMomentObject === true ? value : moment(value);
-
-      setDateRef.current(newDate);
+      if (value._isAMomentObject) {
```
Comment (soupette) [inline packages/buffetjs-custom/src/components/DateTime/index.js:81]: This does not really fix the issue. Maybe we could try a different approach:

```
useEffect(() => {
  if (!isInit) {
    if (!!value && moment(value).isValid()) {
      const newDate = value._isAMomentObject === true ? value : moment(value);

      setTimeStamp(newDate)
   } 

  setInit(true);
 }
 

}, [isInit, value]);


```

Comment (ericliu-el) [reply packages/buffetjs-custom/src/components/DateTime/index.js:81]: Thanks, I think both ways work. Yours is better, then we can remove the `setDateRef` here. I will make an update.

Comment (ericliu-el) [reply packages/buffetjs-custom/src/components/DateTime/index.js]: I have made an update, the test case failed and I made a fix. please check if it works well

**Comment by aksdevac** [comment]
can this be included in next strapi release?

**Pull request merged**